core/src/main/java/org/apache/iceberg/rest/RestCatalog.java

```diff
@@ -0,0 +1,274 @@
+        "Cannot initialize the RestCatalog as the baseUrl is a required parameter.");
+
+    this.baseUrl = properties.get("baseUrl");
+    this.fileIO = initializeFileIO(properties);
+
+    this.mapper = new ObjectMapper();
+    RequestResponseSerializers.registerAll(mapper);
+
+    // TODO - We can possibly handle multiple warehouses via one RestCatalog to reuse the connection pool
+    //        and for cross database calls if users need to authenticate with each.
+    restClient = HttpClient.builder()
+        .baseUrl(String.format("%s/warehouse/%s", baseUrl, catalogName))
+        .mapper(mapper)
+        .defaultErrorHandler(ErrorHandlers.tableErrorHandler())
+        .build();
+  }
+
+  @Override
+  public String name() {
+    return catalogName;
+  }
+
+  @Override
+  public Table createTable(
+      TableIdentifier identifier, Schema schema, PartitionSpec spec, String location,
+      Map<String, String> props) {
+    throw new UnsupportedOperationException("Not implemented: createTable");
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
+    // Users might have not authenticated, possibly need to reauthenticate, or get authentication info per request.
+    //   Though usually, auth tokens are persistent for at least some period of time.
+    return newTableOps(tableIdentifier, null);
+  }
+
+  protected RestTableOperations newTableOps(TableIdentifier tableIdentifier, String authToken) {
+    throw new UnsupportedOperationException("Not implemented: newTableOps");
+  }
+
+  @Override
+  protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
+    throw new UnsupportedOperationException("Not implemented: defaultWarehouseLocation");
+  }
+
+  @Override
+  public List<TableIdentifier> listTables(Namespace namespace) {
+    throw new UnsupportedOperationException("Not implemented: listTables");
+  }
+
+  @Override
+  public boolean tableExists(TableIdentifier identifier) {
+    throw new UnsupportedOperationException("Not implemented: tableExists");
+  }
+
+  @Override
+  public boolean dropTable(TableIdentifier identifier) {
+    return dropTable(identifier, false);
+  }
+
+  @Override
+  public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    throw new UnsupportedOperationException("Not implemented: dropTable");
+  }
+
+  @Override
+  public void renameTable(TableIdentifier from, TableIdentifier to) {
+    throw new UnsupportedOperationException("Not implemented: renameTable");
+  }
+
+  @Override
+  public void close() throws IOException {
+    restClient.close();
+  }
+
+  @Override
+  public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    CreateNamespaceRequest req = CreateNamespaceRequest.builder()
+        .withNamespaceName(namespace.toString())
+        .withProperties(metadata)
+        .build();
+
+    String path = properties.getOrDefault("create-namespace-path", "databases");
```

Comment:
Does the default here need a starting `/`?

Reply:
Hmm. I don't think it does, but I will add in a check.

Even if it doesn't, it seems a bit weird to have a path that doesn't start with a `/`. So I'll add some normalization code (or use some from elsewhere) to allow for that.